src/wristpy/core/computations.py

```diff
@@ -0,0 +1,51 @@
+"""Calculate base metrics, enmo, anglez, and non-wear detection."""
+
+import polars as pl
+
+from wristpy.core.models import Measurement
+
+
+def moving_mean(array: Measurement, epoch_length: int = 5) -> Measurement:
+    """Calculate the moving mean of the sensor data in array.
+
+    Args:
+        array: The Measurement object with the sensor data we want to take the mean of
+        epoch_length: The length, in seconds, of the window.
+
+    Returns:
+        The moving mean of the array in a new Measurement instance.
+
+    Raises:
+        ValueError: If the epoch length is not an integer or is less than 1.
+    """
+    if not isinstance(epoch_length, int):
```

Comment:
not sure if we need the isinstance check, we have the type hinting and this is the kind of function only the developer would use directly right?

Reply:
I can see both ways since a non-int would cause it to crash. I was also debating just doing `int(epoch_length)` any non-int instead of raising an error. Thoughts @ReinderVosDeWael ?

Reply:
People who can't read type hints aren't our problem. If the type hint says it is an int, it's an int.

Reply:
Less tongue in cheek, yeah these kinda checks really should be reserved for places where a non technical user enters the program e.g. where they supply us with a file.

Reply:
Got it. Will make the changes and then I think everything should be good